Fix: termux and internal storage paths issues when relative Path is missing in URI

UnschooledGamer opened the pull request:
# Changes: 

- Stores the `gitRepoDir` in localStorage for later usage in `path.ts`
- Fixes path issues on android internal storage & termux when there is no relative Path in URI
- adds `--port` argument for esbuild serve, usage syntax as "--port=8080"

plugin zip to test:
[dist.zip](https://github.com/user-attachments/files/18374960/dist.zip)

UnschooledGamer committed:
fix: :ambulance: Fixes paths formatting issue when git repo directory is has a Directory above in termux, internal storage

- Stores the `gitRepoDir` in localStorage for later usage in `path.ts`
- Fixes path issues on android internal storage & termux when there is no relative Path in URI
- adds `--port` argument for esbuild serve, usage syntax as "--port=8080"

UnschooledGamer committed:
fix: `/` being including at start Of path

UnschooledGamer committed:
revert: plugin dev version bump back to normal

UnschooledGamer committed:
remov:e leftover console.log from path.ts

dikidjatar commented:
I have tested the changes you submitted, but I encountered some issues:

1. For Termux URI, I received a "Permission Denial" error related to MANAGE_DOCUMENTS.
Error: `Cannot create property 'caller' on string 'java.lang.SecurityException: Permission Denial: writing com.termux.filepicker.TermuxDocumentsProvider uri content://com.termux.documents/tree/%2Fdata%2Fdata%2Fcom.termux%2Ffiles%2Fhome%2Facode/document/%2Fdata%2Fdata%2Fcom.termux%2Ffiles%2Fhome%2Facode%2F.git from pid=31261, uid=10234 requires android.permission.MANAGE_DOCUMENTS, or grantUriPermission()'`
2. For internal storage URIs, it only works if the URI points to the parent folder (e.g., `content://com.android.externalstorage.documents/tree/primary:AcodeProjects`). Accessing subfolders (e.g., `content://com.android.externalstorage.documents/tree/primary:AcodeProjects/Test`) results in the same error.

I’m not sure what the exact issue is, but it’s possible that the generated URI differs between devices or Acode versions.

UnschooledGamer committed:
fix: pathToUri failing to determine base directory in Termux home

This Commit fixes when Uri(i.e home::/data/data/com.termux/files/home/somedir) access is scoped to Termux's Home directory by adding a  Placeholder in `uriToPath` Function

UnschooledGamer commented:
@dikidjatar ,
> 1. For Termux URI, I received a "Permission Denial" error related to MANAGE_DOCUMENTS.
Error: Cannot create property 'caller' on string 'java.lang.SecurityException: Permission Denial: writing com.termux.filepicker.TermuxDocumentsProvider uri content://com.termux.documents/tree/%2Fdata%2Fdata%2Fcom.termux%2Ffiles%2Fhome%2Facode/document/%2Fdata%2Fdata%2Fcom.termux%2Ffiles%2Fhome%2Facode%2F.git from pid=31261, uid=10234 requires android.permission.MANAGE_DOCUMENTS, or grantUriPermission()'

I received this Error while testing & trying But somehow now, I no longer receive that Error again, So I cannot test out.

> 2. For internal storage URIs, it only works if the URI points to the parent folder (e.g., content://com.android.externalstorage.documents/tree/primary:AcodeProjects). Accessing subfolders (e.g., content://com.android.externalstorage.documents/tree/primary:AcodeProjects/Test) results in the same error.

"only works if the URI points to the parent folder (e.g., content://com.android.externalstorage.documents/tree/primary:AcodeProjects)"
Agreed. This will work Only if the app has access to it, If the App Has Access specific to `AcodeProjects/Test` then the above URI would error.

For Example:

If I Do:

1. "Add Path" from File Browser then select `AcodeProjects` and add it to `Acode App`
2. Open the `AcodeProjects` Folder in the `Files` Sidebar it will work.

But If I did:

1. "Add Path" from File Browser then select `AcodeProjects/Test` and add it to `Acode App`
2. Opened `AcodeProject/Test` in the `FIles` Sidebar.

Now, If Acode doesn't have access to `AcodeProject` it won't allow this URI: `content://com.android.external storage.documents/tree/primary:AcodeProjects` (I hope you haven't added `AcodeProjects` folder; it's a fresh, clean state App with only `AcodeProject/Test` Added to the File Browser.)


## Better Solutions I think may work:
1. using `window.addedFolder[0].url` directly If it has `::` then appending relative File Path. If `::` is not present we use whatever `window.addedFolder[0].url` returns and construct the `::` part.
2. We Maybe Might be able to get a working formatted URI from [sdcard Api's formatUri method exposed here](https://github.com/Acode-Foundation/Acode/blob/a8aa7831d3a5dbf7849caf1b437e49a4157facd3/src/plugins/sdcard/www/plugin.js#L17) [Actual Code Defined Here](https://github.com/Acode-Foundation/Acode/blob/a8aa7831d3a5dbf7849caf1b437e49a4157facd3/src/plugins/sdcard/src/android/SDcard.java#L888)

dikidjatar commented:
It works well, but there is one issue when initializing a repository in the Termux home folder—it throws an error.

UnschooledGamer commented:
> It works well, but there is one issue when initializing a repository in the Termux home folder—it throws an error.

Yeah, I've just now checked in Android Emulator. Couldn't check before as I had too many files/folders. That slows down the plugin, That's why had to skip it until now.
